ansible/roles/test/files/ptftests/dip_sip.py

```diff
@@ -0,0 +1,199 @@
+
+Notes:
+    Please check the dip_sip.yml file to see the details of how this test works
+'''
+
+#-------------------------------------------------------------------------------
+# Global imports
+#-------------------------------------------------------------------------------
+
+import logging
+import ptf
+
+from ipaddress import ip_address
+from ptf.base_tests import BaseTest
+
+from ptf.testutils import test_params_get
+from ptf.testutils import simple_udp_packet
+from ptf.testutils import simple_udpv6_packet
+from ptf.testutils import send
+from ptf.testutils import verify_packet_any_port
+
+#-------------------------------------------------------------------------------
+# Testcase
+#-------------------------------------------------------------------------------
+
+class PortLagRouterBasedTest:
+    def __init__(self, dipSipTest):
+        self.test = dipSipTest
+        self.testParams = dipSipTest.test_params
+    #--------------------------------------------------------------------------
+
+    def logParams(self):
+        self.test.log("Destination router mac is:  " + self.dstRouterMac)
+        self.test.log("Destination router ipv4 is: " + self.dstRouterIpv4)
+        self.test.log("Destination router ipv6 is: " + self.dstRouterIpv6)
+
+        self.test.log("Destination host mac is:    " + self.dstHostMac)
+        self.test.log("Destination host ipv4 is:   " + self.dstHostIpv4)
+        self.test.log("Destination host ipv6 is:   " + self.dstHostIpv6)
+
+        self.test.log("Source router mac is:       " + self.srcRouterMac)
+        self.test.log("Source router ipv4 is:      " + self.srcRouterIpv4)
+        self.test.log("Source router ipv6 is:      " + self.srcRouterIpv6)
+
+        self.test.log("Source host mac is:         " + self.srcHostMac)
+        self.test.log("Source host ipv4 is:        " + self.srcHostIpv4)
+        self.test.log("Source host ipv6 is:        " + self.srcHostIpv6)
+
+        self.test.log("Destination port ids is:    " + str([int(portId) for portId in self.dstPortIds]))
+        self.test.log("Source port ids is:         " + str([int(portId) for portId in self.srcPortIds]))
+
+        self.test.log("Packet TTL/HL is:           " + str(self.pktTtlHlim))
+    #--------------------------------------------------------------------------
+
+    def setUpParams(self):
+        self.dstRouterMac = self.testParams['dst_router_mac']
+        self.dstRouterIpv4 = self.testParams['dst_router_ipv4']
+        self.dstRouterIpv6 = self.testParams['dst_router_ipv6']
+
+        self.dstHostMac = self.testParams['dst_host_mac']
+        self.dstHostIpv4 = str(ip_address(unicode(self.testParams['dst_router_ipv4'])) + 1)
+        self.dstHostIpv6 = str(ip_address(unicode(self.testParams['dst_router_ipv6'])) + 1)
+
+        self.srcRouterMac = self.testParams['src_router_mac']
+        self.srcRouterIpv4 = self.testParams['src_router_ipv4']
```

Comment:
Just a note, looks like srcRouterIPvX and srcHostIpvX are not used in the test.

Reply:
@prsunny Correct. It is just for logging purpose.